keps/sig-api-machinery/20190226-network-proxy.md

```diff
@@ -160,11 +132,24 @@ Implies no overlapping (i.e. shared) IPs on the network.
 
 We will run a connectivity proxy inside the master network.
 It could work on either a HTTP Connect mechanism or gRPC.
-We will test HTTP Connect to determine scalability and error handling.
-If no issues with either multi-plexing (to prevent an proliferation of
-connections) or with handling connection errors and terminations, then we will
-go with HTTP Connect. We expect to handle metadata in this case using http 'X' headers.
-Alternately the connectivity proxy can expose a gRPC interface to the KAS.
+For the alpha version we will attempt to get this working with HTTP Connect.
+We will evaluate HTTP Connect for scalability, error handling and traffic types.
+For scalability we will be looking at the number of required open connections.
+Increasing usage of webhooks means we need better than 1 request per connection (multiplexing).
+We also need the tunnel to be tolerant of errors in the requests it is transporting.
+HTTP-Connect only supports HTTP requests and not things like DNS requests.
+We assume that for HTTP URL request,s it will be the proxy which does the DNS lookup.
+However this means that we cannot have the KAS perform a DNS request to then do a follow on request. 
```

Comment:
KAS needs to explicitly `Lookup` then `Dial`.

Reply:
I think this section is trying to say that, since in the HTTP CONNECT method, the proxy gets a URL, the proxy has to do the resolution. @cheftako is this correct? HTTP CONNECT only supports HTTP connections and not other types of network requests?

Reply:
That _was_ my understanding. However https://en.wikipedia.org/wiki/HTTP_tunnel indicates after the http tunnel is established that it should just forward all TCP traffic. I will amend the KEP accordingly.